examples/ipv6only-app/esp32/main/main.cpp

```diff
@@ -0,0 +1,152 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "PigweedLoggerMutex.h"
+#include "RpcService.h"
+#include "freertos/FreeRTOS.h"
+#include "freertos/task.h"
+#include "gdm_wifi_service.h"
+#include "nvs_flash.h"
+#include "pw_rpc/server.h"
+
+#include "lwip/err.h"
+#include "lwip/sockets.h"
+#include "lwip/sys.h"
+
+namespace {
+
+const char * TAG = "ipv6only";
+
+constexpr size_t kUdpBufferSize = 512;
+
+constexpr size_t kRpcStackSizeBytes  = (6 * 1024);
+constexpr uint8_t kRpcTaskPriority   = 5;
+constexpr size_t kTestStackSizeBytes = (8 * 1024);
+constexpr uint8_t kTestTaskPriority  = 5;
+
+TaskHandle_t rpcTaskHandle;
+TaskHandle_t testTaskHandle;
+
+void UdpReceiver(void * pvParameters)
+{
+    int portno;                     // port to listen on
+    struct sockaddr_in6 serveraddr; // server's addr
+    char buf[kUdpBufferSize];       // rx message buf
+    char * hostaddrp;               // dotted decimal host addr string
+    int optval;                     // flag value for setsockopt
+    int n;                          // message byte size
+    int sockfd = 0;
+    unsigned int clientlen;         // byte size of client's address
+    struct sockaddr_in6 clientaddr; // client addr
+
+    while (1)
+    {
+        // Start the udp server after the wifi is connectd.
+        chip::rpc::GDMWifiBase::Instance().BlockUntilWifiConnected();
+        ESP_LOGI(TAG, "UDP server starting");
+
+        portno = 8765;
+        // socket: create the parent socket
+        sockfd = socket(AF_INET6, SOCK_DGRAM, 0);
+        if (sockfd < 0)
+        {
+            ESP_LOGE(TAG, "ERROR opening socket");
+            assert(0);
+            return;
```

Comment:
Anyway this will assert. But still do you want to use `vTaskDelete(NULL);` instead of `return;`?

Reply:
done